test/razorIntegrationTests/jestSetup/vsCodeEnvironment.ts

```diff
@@ -0,0 +1,30 @@
+/*---------------------------------------------------------------------------------------------
+ *  Copyright (c) Microsoft Corporation. All rights reserved.
+ *  Licensed under the MIT License. See License.txt in the project root for license information.
+ *--------------------------------------------------------------------------------------------*/
+
+import * as vscode from 'vscode';
+import { JestEnvironmentConfig, EnvironmentContext } from '@jest/environment';
+import { TestEnvironment } from 'jest-environment-node';
+
+/**
+ * Defines a custom jest environment that allows us to replace vscode module imports with the
+ * instance from the vscode extension in the test runner.
+ */
+class VsCodeEnvironment extends TestEnvironment {
+    constructor(config: JestEnvironmentConfig, context: EnvironmentContext) {
+        super(config, context);
+    }
+
+    public async setup() {
+        await super.setup();
+        this.global.vscode = vscode;
+    }
+
+    public async teardown() {
+        this.global.vscode = {};
+        return await super.teardown();
+    }
+}
+
+module.exports = VsCodeEnvironment;
```

Comment:
same here - you could probably just import the other vscode environment instead of re-defining it.

Reply:
I tried and the relative file in the jest.config works too, so I can remove both files in this folder. Seems like that might be better, in case more mocks are needed in future?

Reply:
yeah, removing is better if its easy